### DISCUSSION TITLE
KPL buttons interacting?

### DISCUSSION THREAD

**Discussion by bobby3572** [TD22057/insteon-mqtt#459] [General]
I have two keypadlincs in my system connected to an Insteon Hub 2. I have installed Insteon-MQTT as a plugin to Home Assistant. Using the command I can control the buttons on the keypadlinc on/off. However if one button is on, then I command a different button on.... the first button LED goes out. When I check, it looks like the corresponding "switch" in Home Assistant is still on, however the LED itself goes out.

Any idea why it would be doing that and how to get the buttons to function independently? It is an 8 button keypadlinc.

**Comment by bobby3572** [comment]
I think it might actually be an issue with hub synchronization.    It seems like the button state changes, but then when the hub sends a command it uses whatever state it thought the buttons were (maybe not the latest if hasn't caught up with everything).    I am thinking maybe I need to force a hub sync after I send any command.    Is there a way to do that?

**Comment by TD22057** [comment]
I assume you mean the modem?  The modem doesn't know or care what state the device is in.  It just sends the commands it's told to.  So there is nothing to sync in the modem as far as device state goes.

**Reply by bobby3572** [reply]
Hey there.    So I am using a hub v2 to control the device, not the PLC.    If I change a button state to ON, it seems to undo whatever the previous button press was.   So turn on button 2...   it comes on.   Then I turn on button 3 - button 2 turns off then button 3 turns on.    etc.     I thought maybe they were cross linked or something so I just factory reset everything and started over....    same issue

**Reply by bobby3572** [reply]
something else to note....    the behaviour I am explaining is just with the LEDs on the KPL buttons.    If you look at the switch states in HA, they correctly show what I set them to and are not changing by themselves.

**Reply by bobby3572** [reply]
a bit more testing....   the only exception is the main load button (ie button 1).    It acts completely like it should and is unaffected by any of the other buttons.

**Reply by chris153002** [reply]
@bobby3572  Have you figured out a solution to this? I am having the same issue.
change a button led from off to on and the others already on, turn off.

**Comment by bobby3572** [comment]
There was an update that came through to the addon which helped for me.    I no longer have that effect of changing one turning off.

I do find you can’t send a bunch of commands to it quickly.   Seems to work better with some space between them.    Plus the actual MQTT commands never seem to work as I expect (could be I just don’t understand what they are doing) so most of the time I just command the items on/off as switches.

I do wish there was a way to reliably separate the switch state of the buttons from the LED state (ie command the LED and the switch independently).    I though with this addon I would be able to do that but haven’t been able to get it to work

**Reply by kpfleming** [reply]
On a KPL, only two of the button *have* a switch state (the ones that control the built-in dimmer or relay). The other buttons *only* have an LED state.
